Applications/Spire/Include/Spire/Ui/DateInputWidget.hpp

```diff
@@ -10,6 +10,12 @@ namespace Spire {
   class DateInputWidget : public QLabel {
     public:
 
+      //! Signal type for date selection changes.
+      /*!
+        \param date The selected date.
+      */
+      using SelectedSignal = Signal<void (boost::posix_time::ptime date)>;
```

Comment:
For a date, can you stick to using ```boost::gregorian::date```?

Reply:
I think that a date is more appropriate, but if I recall originally the requirement was to use a ptime because that's what the APIs interacting with this DateInputWidget would expect.

Reply:
That is entirely possible, but go with ```boost::gregorian::date```.